spec/lib/weby/components/image_component_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+describe ImageComponent do
+  it { should have_setting :repository_id }
+  it { should have_setting :size }
+  it { should have_setting :height }
+  it { should have_setting :width }
+  it { should have_setting :target_type }
+  it { should have_setting :target_id }
+  it { should have_setting :url }
+  it { should have_setting :new_tab }
+  it { should have_setting :html_class }
+
+  it { should validate_presence_of :repository_id }
+
+  it { should allow_value("class").for(:html_class) }
+  it { should_not allow_value("<body></bodyx>").for(:html_class) }
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.